### PR TITLE
chore: drop python 3.7 and use ruff/poetry-check/pyupgrade

### PR DESCRIPTION
**Pull request by efiop**
Starting to deal with isolate a bit more and our lint and ci are pretty slow, so done a little bit of cleanup to speed it up by using ruff and a few more linters. No functional changes.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,30 +1,34 @@
...
-         exclude: '^src/.*_pb2.py$'
-         additional_dependencies: [types-protobuf]
--   repo: https://github.com/codespell-project/codespell
-    rev: v2.1.0
-    hooks:
-    -   id: codespell
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: 'v0.3.4'
+  hooks:
+    - id: ruff
```
Comment (squat) [inline .pre-commit-config.yaml:5]: amen, I use ruff everywhere! it's faster and better

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -52,11 +52,10 @@ jobs:
...
     - name: Install uv
-      if: ${{ matrix.python != '3.7' }}
       run: |
         python -m pip install uv
 
     - name: Test
       run: |
         export ISOLATE_PYENV_EXECUTABLE=pyenv/bin/pyenv
-        python -m pytest -vvv
+        python -m pytest -vvv -n auto
```
Comment (efiop) [inline .github/workflows/test.yml:61]: Looks like some of our stuff is running into each other, will do that in a separate PR.

**Comment by efiop** [comment]
Let me see if I can bring back xdist, it cut time by half but was causing some issues in our tests.

File: tests/test_connections.py [hunk 3]
```diff
@@ -159,7 +159,7 @@ def foo():
                 return c
 
             def bar():
-                a = str() + str()
+                a = "" + ""  # noqa: F841
```
Comment (efiop) [inline tests/test_connections.py:162]: This str -> "" change shouldn't matter as we are testing tracebacks here.

**Review by chamini2 — APPROVED** [review]
LGTM
